dlib/type_safe_union/type_safe_union_kernel_abstract.h

```diff
@@ -192,11 +193,13 @@ namespace dlib
         );
         /*!
            ensures
-              - if (T is the same type as one of the template arguments) then
-                 - returns a number indicating which template argument it is. In particular,
-                   if it's the first template argument it returns 1, if the second then 2, and so on.
-               - else
-                  - returns -1
+                - if (T is the same type as one of the template arguments) then
+                    returns a number indicating which template argument it is. In particular,
```

Comment:
Keep the leading `-` for each bullet since that's the style in the rest of the library.